api/envoy/extensions/filters/http/jwt_authn/v3/config.proto

```diff
@@ -119,6 +119,31 @@ message JwtProvider {
   //
   type.matcher.v3.StringMatcher subjects = 19;
 
+  oneof lifetime_constraint {
+    // Restrict the maximum remaining lifetime of a credential from the JwtProvider. For instance, to make sure that tokens
+    // are not valid for more than one hour. Credentials valid for longer will fail validation. Setting this to inf will
+    // require that an expiration is set in the JWT, but not check the lifetime.
```

Comment:
> Setting this to inf will require that an expiration is set in the JWT, but not check the lifetime.

I'm not clear on what this means.  It sounds like the JWT token can have an expiration, and you can also set a lifetime here, and if you set a lifetime here w/ max_lifetime, then it'll override the JWT token expiration, but if you set max_lifetime to `inf`, then the JWT token expiration will be used instead?

Reply:
Let me rephrase that. What we're trying to capture is that if you set a max_lifetime of 24 hours, then tokens that are valid for longer than that will be rejected. This is to stop the case where an issuer issues tokens that are valid for a long period and don't have a way to revoke them if they're stolen.

But, the JWT-SVID [spec](https://github.com/spiffe/spiffe/blob/main/standards/JWT-SVID.md#33-expiration-time) only requires that there is an expiration, but doesn't require a specific value. So the idea is to let people configure their proxies to meet the minimum of the spec or be more strict.

Reply:
Redid the docs